docs/rfcs/2022-12-09-ephemeral-conversations.md

```diff
@@ -0,0 +1,208 @@
+# Ephemeral conversations with existing contacts
+
+Ephemeral conversation inside existing conversation with stricter security properties.
+
+- additional level of encryption for message bodies & text, chat items content & text (more?) inside chat db
+
+- ephemeral conversation key not persisted - only stored in-memory of current chat session, cleared on exiting to background
+
+- separate tables for chat items and messages to avoid gaps in ids? TBC db pages on deletion
+
+- don't persist chat items and messages at all? if yes - how to support multiple ephemeral conversations with different contacts in the same session - holding chat items in memory may become expensive. though only "not yet seen" items may have to be held in memory - after opening ephemeral conversation no longer keep them in memory; in this case closing ephemeral conversation screen (not fully exiting but keeping it in list of current ephemeral conversations) and re-opening also does not restore chat items, though connection and key are preserved
+
+- if multiple ephemeral conversations are allowed - how to know they have new messages - should there be notifications for them? only local or push notifications too? not a regular notification but indication in chat list?
+
+- disabled features? e.g. "reply" if messages aren't persisted. voice messages, files, etc.? if files are supported they are deleted exiting, should also be a part of chat start cleanup process (see below)
+
+- contact is required to be verified to start ephemeral conversation - improves guarantee that the key for ephemeral conversation is agreed in a secure context
+
+- no visibility of contact profile in UI
+
+- separated with a blank screen / transition from a main conversation to prevent them appearing in the same screen
+
+- new entity - not a contact?
```

Comment:
we can just use incognito contact for that with some reference to the original contact